scripts/superdesk/itemList/itemList.js

```diff
@@ -5,6 +5,74 @@ var DEFAULT_OPTIONS = {
     sort: [{_updated: 'desc'}]
 };
 
+/**
+ * Common part for sdItemListWidget and sdRelatedItemListWidget directives.
+ *
+ * Params:
+ * @param {Object} scope
+ * @param {Object} itemList
+ * @param {Object} itemPinService
+ */
+function extendScope(scope, itemList, itemPinService) {
+    scope.view = function(item) {
+        scope.selected = item;
+    };
+
+    scope.toggleItemType = function(itemType) {
+        if (scope.itemListOptions.types.indexOf(itemType) > -1) {
+            scope.itemListOptions.types = _.without(scope.itemListOptions.types, itemType);
+        } else {
+            scope.itemListOptions.types.push(itemType);
+        }
+    };
+
+    scope.isItemTypeEnabled = function(itemType) {
+        return scope.itemListOptions.types.indexOf(itemType) > -1;
+    };
+
+    scope.pin = function(item) {
+        itemPinService.add(scope.options.pinMode, _.clone(item));
+    };
+
+    scope.unpin = function(item) {
+        itemPinService.remove(item);
+    };
+
+    scope.isPinned = function(item) {
```

Comment:
You don't have to change this now, but it's worth noting for the future that functions like this serve no purpose being on the `scope` and generally the scope should be limited to items that you wish to (two-way) bind to, such as a model or in rare cases functions. If my understanding of the way Angular works is correct, this affects performance seriously and all these scope items become processed during the digest cycle.

This is one of the main reasons people run into performance issues with Angular. A better approach is to create controllers and have these functions there.

It's ok to leave this as is for now, but I'm just saying this as a remark since you were saying that you are learning Angular.

You don't need to take any action.